renn/data/tokenizers.py

```diff
@@ -15,6 +18,114 @@
 UNK = '<unk>'
 CLS = '<cls>'
 SEP = '<sep>'
+EOS = '<eos>'
+BOS = '<bos>'
+
+
+def _get_transform_fn(lower: bool, separate_punctuation: bool):
```

Comment:
This way of generating a transform function is a bit unwieldy, it won't scale well as we add new options (the number of branches grow exponentially).

Instead, my preferences is to remove this helper function and instead let the user define and pass in their own transform function to `text_generator` below (and we can default it to something sensible).

Reply:
Agreed.  I took out  `_get_transform_fn`, and have a simple function called `lowercase_strip()` which can be used as a default.